## Extracción de muertes por Provincia

In [35]:
import requests
import pandas as pd

url = "https://covid-19-statistics.p.rapidapi.com/regions"

headers = {
	"X-RapidAPI-Key": "75e6f2eaf6msh1c8c0c7d9dc6d9cp15eb67jsnf082524e4c2a",
	"X-RapidAPI-Host": "covid-19-statistics.p.rapidapi.com"
}

response = requests.request("GET", url, headers=headers)

data = response.json()
pd.DataFrame(data['data'])

iso                 name
0    CHN                China
1    TWN  Taipei and environs
2    USA                   US
3    JPN                Japan
4    THA             Thailand
..   ...                  ...
214  KIR             Kiribati
215  PLW                Palau
216  TON                Tonga
217  NRU                Nauru
218  TUV               Tuvalu

[219 rows x 2 columns]

In [37]:
import requests
import pandas as pd

url = "https://covid-19-statistics.p.rapidapi.com/provinces"

querystring = {"iso":"CHN"}

headers = {
	"X-RapidAPI-Key": "75e6f2eaf6msh1c8c0c7d9dc6d9cp15eb67jsnf082524e4c2a",
	"X-RapidAPI-Host": "covid-19-statistics.p.rapidapi.com"
}

response = requests.request("GET", url, headers=headers, params=querystring)

data = response.json()
pd.DataFrame(data['data'])

iso   name        province      lat      long
0   CHN  China           Anhui  31.8257  117.2264
1   CHN  China         Beijing  40.1824  116.4142
2   CHN  China       Chongqing  30.0572  107.8740
3   CHN  China          Fujian  26.0789  117.9874
4   CHN  China           Gansu  36.0611  103.8343
5   CHN  China       Guangdong  23.3417  113.4244
6   CHN  China         Guangxi  23.8298  108.7881
7   CHN  China         Guizhou  26.8154  106.8748
8   CHN  China          Hainan  19.1959  109.7453
9   CHN  China           Hebei  38.0428  114.5149
10  CHN  China    Heilongjiang  47.8620  127.7615
11  CHN  China           Henan  33.8820  113.6140
12  CHN  China       Hong Kong  22.3000  114.2000
13  CHN  China   Hong Kong SAR     22.3     114.2
14  CHN  China           Hubei  30.9756  112.2707
15  CHN  China           Hunan  27.6104  111.7088
16  CHN  China  Inner Mongolia  44.0935  113.9448
17  CHN  China         Jiangsu  32.9711  119.4550
18  CHN  China         Jiangxi  27.6140  115.7221
19  CHN  China           Jilin  43.6661  126.1923
20  CHN  China        Liaoning  41.2956  122.6085
21  CHN  China           Macau  22.1667  113.5500
22  CHN  China       Macau SAR  22.1667    113.55
23  CHN  China         Ningxia  37.2692  106.1655
24  CHN  China         Qinghai  35.7452   95.9956
25  CHN  China         Shaanxi  35.1917  108.8701
26  CHN  China        Shandong  36.3427  118.1498
27  CHN  China        Shanghai  31.2020  121.4491
28  CHN  China          Shanxi  37.5777  112.2922
29  CHN  China         Sichuan  30.6171  102.7103
30  CHN  China         Tianjin  39.3054  117.3230
31  CHN  China           Tibet  31.6927   88.0924
32  CHN  China         Unknown     None      None
33  CHN  China        Xinjiang  41.1129   85.2401
34  CHN  China          Yunnan  24.9740  101.4870
35  CHN  China        Zhejiang  29.1832  120.0934

In [57]:
import requests
#extraemos los reportes totales del 2020-02-01 
url = "https://covid-19-statistics.p.rapidapi.com/reports/total"

querystring = {"date":"2020-02-01"}

headers = {
	"X-RapidAPI-Key": "521a9a1cb9mshb2ed089bbc4e3cbp1b8bfbjsn1a39d05b0d58",
	"X-RapidAPI-Host": "covid-19-statistics.p.rapidapi.com"
}

response = requests.request("GET", url, headers=headers, params=querystring)

data = response.json()
df = pd.DataFrame(data['data'], index=[0])
df

date          last_update  confirmed  confirmed_diff  deaths  \
0  2020-02-01  2020-02-01 19:53:00      12038            2106     259   

   deaths_diff  recovered  recovered_diff  active  active_diff  fatality_rate  
0           46        284              62   11495         1998         0.0215

In [84]:
import requests
#extraemos 
url = "https://covid-19-statistics.p.rapidapi.com/reports"

querystring = {"city_name":"Autauga","region_province":"Alabama","iso":"USA","region_name":"US","q":"US Alabama","date":"2020-04-16"}

headers = {
	"X-RapidAPI-Key": "75e6f2eaf6msh1c8c0c7d9dc6d9cp15eb67jsnf082524e4c2a",
	"X-RapidAPI-Host": "covid-19-statistics.p.rapidapi.com"
}

response = requests.request("GET", url, headers=headers, params=querystring)

data = response.json()
df1 = pd.DataFrame(data['data'], columns=['date','confirmed','deaths','recovered'])
df2 = pd.DataFrame([data['data'][0]['region']], columns=['iso','name','province','lat','long'])
df3 = pd.DataFrame(data['data'][0]['region']['cities'])

In [85]:
df1

date  confirmed  deaths  recovered
0  2020-04-16       4345     133          0

In [86]:
df2

iso name province      lat      long
0  USA   US  Alabama  32.3182  -86.9023

In [87]:
df3

name        date  fips          lat          long  confirmed  deaths  \
0  Autauga  2020-04-16  1001  32.53952745  -86.64408227         26       1   

   confirmed_diff  deaths_diff          last_update  
0               2            0  2020-04-16 23:30:51

In [52]:
import requests
#extraemos los reportes totales del 2020-04-07
url = "https://covid-19-statistics.p.rapidapi.com/reports/total"

querystring = {"date":"2020-04-07"}

headers = {
	"X-RapidAPI-Key": "75e6f2eaf6msh1c8c0c7d9dc6d9cp15eb67jsnf082524e4c2a",
	"X-RapidAPI-Host": "covid-19-statistics.p.rapidapi.com"
}

response = requests.request("GET", url, headers=headers, params=querystring)

data = response.json()
df = pd.DataFrame(data['data'], index=[0])
df

date          last_update  confirmed  confirmed_diff  deaths  \
0  2020-04-07  2020-04-07 23:11:31    1426096           80995   81865   

   deaths_diff  recovered  recovered_diff   active  active_diff  fatality_rate  
0         7300     300054           23539  1044177        50156         0.0574

#### En 2020-04-07 hibieron  1426096  casos cofirmados, 81865 fallecidos y con 1044177 casos activos.